### PR TITLE
Avoid uninteresting calls google test warnings

### PR DESCRIPTION
**Pull request by ghost**
This PR is **ready** for review.

### Risk
This PR makes **no** API changes.

### Summary
This PR provided fix of google tests warnings in the next UT's:

- rc_commands_test
- message_helper_test
- help_prompt_manager_test
- application_manager_impl_mock_hmi_test
- policy_manager_test
- connection_handler_test
- transport_manager_test
- data_resumption_test
- sdl_commands_test

**Was fixed for PROPRIETARY & EXTERNAL_PROPRIETARY mode**

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

File: src/components/application_manager/rpc_plugins/rc_rpc_plugin/test/commands/button_press_request_test.cc [hunk 1]
```diff
@@ -126,10 +126,10 @@ class ButtonPressRequestTest
     ON_CALL(mock_hmi_capabilities_, rc_capability())
         .WillByDefault(Return(&rc_capabilities_));
     ON_CALL(*mock_app_, policy_app_id()).WillByDefault(Return(kPolicyAppId));
-    ON_CALL(mock_policy_handler_,
-            CheckHMIType(kPolicyAppId,
-                         mobile_apis::AppHMIType::eType::REMOTE_CONTROL,
-                         nullptr)).WillByDefault(Return(true));
+    EXPECT_CALL(mock_policy_handler_,
```
Comment (LuxoftAKutsan) [inline src/components/application_manager/rpc_plugins/rc_rpc_plugin/test/commands/button_press_request_test.cc]: @ValeriiMalkov Expects in Setup smells. Are you testing calling `CheckHMIType` in all of tests in this files?

**Review by LuxoftAKutsan — COMMENTED** [review]
EXPECT_CALL should be used in case if it is **part** of test case, I mean if calling of this method is actually behavior that certain test cases is testing.  
For providing required for code data, we should use ON_CALLs; 

### Example : 
**Code:**
```cpp
void ConnectToServer() {
   credantials = profile.GetCredentials();
   if (!credantials) {
     return;
   }
   io_device = server.Connect(credantials);
   ok = logic.OnServerConnected(io_device)
   if (ok) {
      observer.Notify()
   }
}
```

**Good tests:**
```cpp
TEST(ConnectToServer, EmptyCredentials)  {
  Nice<profile_mock>;
  server_mock;
  ON_CALL(profile_mock, GetCredentials):Return(EMPTY);
  EXPECT_CALL(server_mock,Connect):Times(0) ;
  ConnectToServer();
}

TEST(ConnectToServer, ExpectConnection)  {
  Nice<profile_mock>;
  server_mock;
  Nice<logic_mock>;
  Nice<observer_mock>; 
  ON_CALL(profile_mock, GetCredentials):Return("some credentials");
  ON_CALL(logic_mock, OnServerConnected(some_device)):Return(ok);
  EXPECT_CALL(server_mock, Connect("some credentials")):Return(some_device);
  ConnectToServer();
}

TEST(ConnectToServer, NotifyServerConnected)  {
  Nice<profile_mock>;
  Nice<server_mock>;
  logic_mock;
  Nice<observer_mock>; 
  ON_CALL(profile_mock, GetCredentials):Return("some credentials");
  ON_CALL(server_mock, Connect("some credentials")):Return(some_device)
  EXPECT_CALL(logic_mock, OnServerConnected(some_device)):Return(ok);
  ConnectToServer();
}


TEST(ConnectToServer, NotifyConnectionSuccesfulProcessed)  {
  Nice<profile_mock>;
  Nice<server_mock>;
  Nice<logic_mock>;
  observer_mock; 
  ON_CALL(profile_mock, GetCredentials):Return("some credentials");
  ON_CALL(server_mock, Connect("some credentials")):Return(some_device)
  ON_CALL(logic_mock, OnServerConnected(some_device)):Return(ok);
  EXPECT_CALL(observer_mock, Notify);
  ConnectToServer();
}

TEST(ConnectToServer, ConnectionNotProcessedMissedNotification)  {
  Nice<profile_mock>;
  Nice<server_mock>;
  Nice<logic_mock>;
  observer_mock; 
  ON_CALL(profile_mock, GetCredentials):Return("some credentials");
  ON_CALL(server_mock, Connect("some credentials")):Return(some_device)
  ON_CALL(logic_mock, OnServerConnected(some_device)):Return(not ok);
  EXPECT_CALL(observer_mock, Notify):Times(0);
  ConnectToServer();
}
```

Please review usage of ON and EXPECT calls!

File: src/components/application_manager/rpc_plugins/rc_rpc_plugin/test/commands/button_press_request_test.cc (hunk 1 above)
Comment (ghost) [reply src/components/application_manager/rpc_plugins/rc_rpc_plugin/test/commands/button_press_request_test.cc]: @LuxoftAKutsan Was fixed and forced updated. Please review changes.

File: src/components/application_manager/test/application_manager_impl_mock_hmi_test.cc [hunk 2]
```diff
@@ -202,8 +202,8 @@ TEST_F(ApplicationManagerImplMockHmiTest,
 
   EXPECT_CALL(*mock_rpc_plugin_manager, FindPluginToProcess(_, _))
       .WillRepeatedly(Return(mock_rpc_plugin_opt));
-  ON_CALL(mock_rpc_plugin, GetCommandFactory())
-      .WillByDefault(ReturnRef(mock_command_factory));
+  EXPECT_CALL(mock_rpc_plugin, GetCommandFactory())
```
Comment (LuxoftAKutsan) [inline src/components/application_manager/test/application_manager_impl_mock_hmi_test.cc]: @ValeriiMalkov why expect?

File: src/components/connection_handler/test/connection_handler_impl_test.cc [hunk 3]
```diff
@@ -808,6 +808,9 @@ TEST_F(ConnectionHandlerTest, OnDeviceRemoved_ServiceNotStarted) {
   EXPECT_CALL(mock_connection_handler_observer, RemoveDevice(dev_handle1));
   EXPECT_CALL(mock_connection_handler_observer, OnServiceEndedCallback(_, _, _))
       .Times(0);
+  EXPECT_CALL(
```
Comment (LuxoftAKutsan) [inline src/components/connection_handler/test/connection_handler_impl_test.cc]: @ValeriiMalkov why expect ?

File: src/components/transport_manager/test/transport_manager_impl_test.cc [hunk 4]
```diff
@@ -360,6 +361,7 @@ TEST_F(TransportManagerImplTest, ConnectDevice) {
   HandleDeviceListUpdated();
   EXPECT_CALL(*mock_adapter_, ConnectDevice(mac_address_))
       .WillOnce(Return(TransportAdapter::OK));
+
```
Comment (LuxoftAKutsan) [inline src/components/transport_manager/test/transport_manager_impl_test.cc]: @ValeriiMalkov empty line required ?

File: src/components/transport_manager/test/transport_manager_impl_test.cc [hunk 5]
```diff
@@ -746,6 +748,7 @@ TEST_F(TransportManagerImplTest, ReceiveEventFromDevice_DeviceListUpdated) {
       .WillOnce(NotifyTestAsyncWaiter(&waiter));
   EXPECT_CALL(*tm_listener_, OnDeviceAdded(dev_info_))
       .WillOnce(NotifyTestAsyncWaiter(&waiter));
+  EXPECT_CALL(*tm_listener_, OnDeviceListUpdated(_));
```
Comment (LuxoftAKutsan) [inline src/components/transport_manager/test/transport_manager_impl_test.cc]: @ValeriiMalkov please check internal of the update

**Review by LuxoftAKutsan — CHANGES_REQUESTED** [review]

File: src/components/application_manager/test/application_manager_impl_mock_hmi_test.cc (hunk 2 above)
Comment (ghost) [reply src/components/application_manager/test/application_manager_impl_mock_hmi_test.cc]: @LuxoftAKutsan Was returned to ON_CALL and force updated

File: src/components/connection_handler/test/connection_handler_impl_test.cc (hunk 3 above)
Comment (ghost) [reply src/components/connection_handler/test/connection_handler_impl_test.cc]: @LuxoftAKutsan Was returned to ON_CALL and force updated

File: src/components/transport_manager/test/transport_manager_impl_test.cc (hunk 4 above)
Comment (ghost) [reply src/components/transport_manager/test/transport_manager_impl_test.cc]: @LuxoftAKutsan Not required. Removed and force updated

File: src/components/transport_manager/test/transport_manager_impl_test.cc (hunk 5 above)
Comment (ghost) [reply src/components/transport_manager/test/transport_manager_impl_test.cc]: @LuxoftAKutsan Arg of function was added

**Review by SNiukalov — APPROVED** [review]

File: src/components/application_manager/test/message_helper/message_helper_test.cc [hunk 6]
```diff
@@ -593,7 +594,8 @@ class MessageHelperTest : public ::testing::Test {
       , delta_from_functions_id(32768) {}
 
  protected:
-  application_manager_test::MockApplicationManager mock_application_manager;
+  testing::NiceMock<application_manager_test::MockApplicationManager>
+      mock_application_manager;
```
Comment (AByzhynar) [inline src/components/application_manager/test/message_helper/message_helper_test.cc:598]: @ValeriiMalkov 
- named wrongly according to SDL Coding Style

File: src/components/application_manager/rpc_plugins/rc_rpc_plugin/test/commands/rc_get_interior_vehicle_data_consent_test.cc [hunk 7]
```diff
@@ -174,7 +174,7 @@ class RCGetInteriorVehicleDataConsentTest
 
  protected:
   std::shared_ptr<MockApplication> mock_app_;
-  MockRequestControlerSettings mock_request_controler;
+  NiceMock<MockRequestControlerSettings> mock_request_controler;
```
Comment (AByzhynar) [inline src/components/application_manager/rpc_plugins/rc_rpc_plugin/test/commands/rc_get_interior_vehicle_data_consent_test.cc:177]: @ValeriiMalkov 
- named wrongly according to SDL Coding Style

**Review by AByzhynar — CHANGES_REQUESTED** [review]

**Review by LuxoftAKutsan — APPROVED** [review]

**Review by BSolonenko — APPROVED** [review]

**Comment by ghost** [comment]
PR moved to SDLMirror

**Pull request closed without merging**